lib/ansible/modules/net_tools/haproxy.py

```diff
@@ -389,6 +394,15 @@ def drain(self, host, backend, status='DRAIN'):
             cmd = "set server $pxname/$svname state drain"
             self.execute_for_backends(cmd, backend, host, status)
 
+    def disable_after_drain(self, pxname, svname):
+        self.wait = False
+        self.disabled(self.host, self.backend, self.shutdown_sessions)
+        state = self.get_state_for(pxname, svname)
+        if state[0]["status"] == "MAINT":
```

Comment:
`wait_until_status` method returns the returned value:
https://github.com/ansible/ansible/blob/34242b928f2921df0ba174025f8bff5c9140f79b/lib/ansible/modules/net_tools/haproxy.py#L357
but caller (`execute_for_backends`) don't use it:
https://github.com/ansible/ansible/blob/34242b928f2921df0ba174025f8bff5c9140f79b/lib/ansible/modules/net_tools/haproxy.py#L320
Isn't that a problem ?

Reply:
I added an error if we can't manage to reach the status expected